### DISCUSSION TITLE
Multi-object Tracking Datasets Overview - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#3648] [Docs]
# Multi-object Tracking Datasets Overview - Ultralytics YOLOv8 Docs

Discover the datasets compatible with Multi-Object Detector. Train your trackers and make your detections more efficient with Ultralytics' YOLO.

https://docs.ultralytics.com/datasets/track/

**Comment by kibriadar** [comment]
Hi,
 quick query. If i understand it right, one would be able to train object trackers on their custom dataset?

**Reply by glenn-jocher** [reply]
@kibriadar you would train a normal Detect, Segment or Pose model, and then you can use any of these for tracking with i.e. `yolo track model=...` :)

**Comment by MarcosFdzRdz** [comment]
So, how can I add my own dataset for training? I mean, I would like to train it from scratch and/or maybe use my own data for fine tunning. Am I able to do it with YOLOv8? For example for segmentation

**Reply by MarcosFdzRdz** [reply]
I have found this: https://docs.ultralytics.com/datasets/segment/
But I still having doubts, I mean, is it supposed that I should annotate the "xn, yn" coordinates in order to define segmentation mask? Maybe I'm lucky enough and someone already made a code to adapt semantic segmentation masks to this format?
Then if I'm not wrong, the thing is:
1. Define .yaml file with classes and paths for train, validation and test sets.
2. Generate 1 .txt file for each image with the format, row by row for every object <class-index> <x1> <y1> <x2> <y2> ...
3. And then run normally the model with the yolo train command, with data=mydataset.yaml?

Am I right?

**Reply by glenn-jocher** [reply]
@MarcosFdzRdz yes, you're on the right track! To train YOLOv8 on your own dataset for segmentation, you'll need to prepare your data accordingly:

1. Create a `.yaml` file that specifies your classes and the paths to your train, validation, and test sets.
2. Annotate your images for segmentation by creating corresponding `.txt` files for each image. These should contain one row per object, with the format `<class-index> <x1> <y1> <x2> <y2> ...`, where `xn, yn` are the coordinates that define the segmentation mask.
3. Once your dataset is prepared and your `.yaml` file is set up, you can proceed to train your model using the YOLOv8 train command with `data=mydataset.yaml`.

If you're looking for a tool to help convert semantic segmentation masks to the required format, you might want to check out existing open-source projects or consider writing a custom script to automate this process. The community often shares utilities for such tasks, so it's worth exploring or asking in the community forums.

After these steps, you should be ready to train your model on your custom segmentation dataset. Good luck with your training! 🚀
